src/components/package-port/CodeAndPreview.tsx

```diff
@@ -346,7 +308,16 @@ export function CodeAndPreview({ pkg }: Props) {
     })
   }
 
-  const { handleCreateFile } = useFileManagement(state, setState)
+  const { handleCreateFile, handleDeleteFile, handleSave, fsMap } =
+    useFileManagement({
+      setCodeAndPreviewState: setState,
+      pkg,
```

Comment:
pass the package_id to give this hook a good api. `useQuery` knows how to use cached results. You can also pass nothing at all, since `useCurrentPackageId` will handle loading the current package